### PR TITLE
[chalf] rand, randn, full, empty, ones, zeros

### PR DESCRIPTION
**Pull request by kshitij12345**
Ref: https://github.com/pytorch/pytorch/issues/74537

### CODE REVIEW

File: test/test_tensor_creation_ops.py [hunk 1]
```diff
@@ -1968,11 +1968,12 @@ def test_empty_full(self, device):
         torch_device = torch.device(device)
         device_type = torch_device.type
 
+        dtypes = get_all_dtypes(include_half=False, include_bfloat16=False, include_complex32=True)
         if device_type == 'cpu':
-            do_test_empty_full(self, get_all_math_dtypes('cpu'), torch.strided, torch_device)
+            do_test_empty_full(self, dtypes, torch.strided, torch_device)
         if device_type == 'cuda':
-            do_test_empty_full(self, get_all_math_dtypes('cpu'), torch.strided, None)
```
Comment (kshitij12345) [inline test/test_tensor_creation_ops.py:1974]: Any idea if this was intended to be `cpu` even when `device_type` is `cuda`

cc: @mruberry

Comment (kshitij12345) [reply test/test_tensor_creation_ops.py:1974]: Ref PR: https://github.com/pytorch/pytorch/pull/43104/files

**Review by anjali411 — APPROVED** [review]

File: test/test_tensor_creation_ops.py (hunk 1 above)
Comment (mruberry) [reply test/test_tensor_creation_ops.py:1974]: When refactoring our test suite I often did the minimum to get the tests to work in the new framework (it was a lot of LoC!), so I probably just didn't update this test completely. It doesn't even use the @dtypes decorator!

(or maybe the @dtypes decorator didn't even exist at the time)

**Comment by kshitij12345** [comment]
Failure on ROCm looks irrelevant

https://github.com/pytorch/pytorch/runs/6378177208?check_suite_focus=true#step:7:46700

```
======================================================================
[46701](https://github.com/pytorch/pytorch/runs/6378177208?check_suite_focus=true#step:7:46701)
ERROR [0.462s]: test_broadcast_in_dim_cuda_float32 (__main__.TestPrimsCUDA)
[46702](https://github.com/pytorch/pytorch/runs/6378177208?check_suite_focus=true#step:7:46702)
----------------------------------------------------------------------
[46703](https://github.com/pytorch/pytorch/runs/6378177208?check_suite_focus=true#step:7:46703)
Traceback (most recent call last):
[46704](https://github.com/pytorch/pytorch/runs/6378177208?check_suite_focus=true#step:7:46704)
  File "/opt/conda/lib/python3.7/site-packages/torch/_prims/executor.py", line 64, in execute
[46705](https://github.com/pytorch/pytorch/runs/6378177208?check_suite_focus=true#step:7:46705)
    tuple(arg for arg in args if isinstance(arg, torch.Tensor))
[46706](https://github.com/pytorch/pytorch/runs/6378177208?check_suite_focus=true#step:7:46706)
RuntimeError: false INTERNAL ASSERT FAILED at "/var/lib/jenkins/workspace/torch/csrc/jit/codegen/cuda/executor_utils.cpp":1054, please report a bug to PyTorch. namespace CudaCodeGen {
```

**Pull request closed without merging**